README.md

```diff
@@ -211,42 +212,163 @@ val records = sc.newAPIHadoopFile(
- sc.hadoopConfiguration.set("fs.s3a.secret.key", "YOUR_SECRET_ACCESS_KEY")
- ```
-
- Python users will have to use a slightly different method to modify the `hadoopConfiguration`, since this field is not exposed in all versions of PySpark. Although the following command relies on some Spark internals, it should work with all PySpark versions and is unlikely to break or change in the future:
-
- ```python
- sc._jsc.hadoopConfiguration().set("fs.s3n.awsAccessKeyId", "YOUR_KEY_ID")
- sc._jsc.hadoopConfiguration().set("fs.s3n.awsSecretAccessKey", "YOUR_SECRET_ACCESS_KEY")
- ```
+```
+                            ┌───────┐
+       ┌───────────────────▶│  S3   │◀─────────────────┐
+       │    IAM or keys     └───────┘    IAM or keys   │
+       │                        ▲                      │
+       │                        │ IAM or keys          │
+       ▼                        ▼               ┌──────▼────┐
+┌────────────┐            ┌───────────┐         │┌──────────┴┐
+│  Redshift  │            │   Spark   │         ││   Spark   │
+│            │◀──────────▶│  Driver   │◀────────▶┤ Executors │
+└────────────┘            └───────────┘          └───────────┘
+               JDBC with                  Configured
+               username /                     in
+                password                    Spark
+            (can enable SSL)
+```
 
-3. **Encode keys in `tempdir` URI**: For example, the URI `s3n://ACCESSKEY:SECRETKEY@bucket/path/to/temp/dir` encodes the key pair (`ACCESSKEY`, `SECRETKEY`). Due to [Hadoop limitations](https://issues.apache.org/jira/browse/HADOOP-3733), this approach will not work for secret keys which contain forward slash (`/`) characters.
-4. **Set the `aws_iam_role` parameter:** If set, this takes precedence over any other authentication option.  You will need to have this IAM role attached to the Redshift cluster which allows read/write access to your `tempdir` bucket.  More info [here](http://docs.aws.amazon.com/redshift/latest/mgmt/copy-unload-iam-role.html)
-5. **IAM instance profiles:** If you are running on EC2 and authenticate to S3 using IAM and [instance profiles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html), then you must must configure the `temporary_aws_access_key_id`, `temporary_aws_secret_access_key`, and `temporary_aws_session_token` configuration properties to point to temporary keys created via the AWS [Security Token Service](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html). These temporary keys will then be passed to Redshift via `LOAD` and `UNLOAD` commands.
+This library reads and writes data to S3 when transferring data to/from Redshift. As a result, it
+requires AWS credentials with read and write access to a S3 bucket (specified using the `tempdir`
+configuration parameter).
+
+> **:warning: Note**: This library does not clean up the temporary files that it creates in S3.
```

Comment:
These are the temporary files being referred to in the _Encrypting `UNLOAD` data stored in S3 (data stored when reading from Redshift)_ and _Encrypting `COPY` data stored in S3 (data stored when writing to Redshift)_ headings under the _Encryption_ setting.